test/integration/api/shutdown.tap.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2021 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const nock = require('nock')
+
+const helper = require('../../lib/agent_helper')
+const API = require('../../../api')
+
+// This key is hardcoded in the agent helper
+const EXPECTED_LICENSE_KEY = 'license key here'
+const TEST_DOMAIN = 'test-collector.newrelic.com'
+const TEST_COLLECTOR_URL = `https://${TEST_DOMAIN}`
+
+// TODO: should work after the agent has restarted
+tap.test('#shutdown', (t) => {
+  t.autoend()
+
+  let agent = null
+  let api = null
+
+  t.beforeEach(() => {
+    nock.disableNetConnect()
+
+    agent = helper.loadMockedAgent({
+      license_key: EXPECTED_LICENSE_KEY,
+      host: TEST_DOMAIN
+    })
+
+    agent.config.no_immediate_harvest = true
+
+    api = new API(agent)
+  })
+
+  t.afterEach((t) => {
+    helper.unloadAgent(agent)
+
+    if (!nock.isDone()) {
+      /* eslint-disable no-console */
+      console.error('Cleaning pending mocks: %j', nock.pendingMocks())
+      /* eslint-enable no-console */
+      nock.cleanAll()
+
+      t.fail('Failed to hit all expected endpoints.')
+    }
+
+    nock.enableNetConnect()
+  })
+
+  t.test('should force harvest and callback after agent restart', (t) => {
+    setupConnectionEndpoints('run-id-1')
+    agent.start((error) => {
+      t.error(error)
+
+      setupConnectionEndpoints('run-id-2')
+      agent.collector.restart(() => {
+        setupShutdownEndpoints('run-id-2')
+        api.shutdown({ collectPendingData: true }, (error) => {
```

Comment:
The issue being fixed is explicitly for the case where `collectPendingData` is true. Otherwise, the agent state of 'started' doesn't matter.

I'm all for additional test coverage here, just wasn't the focus of this PR.

Reply:
yea that was my only thought, just while we're in here adding some missing shutdown scenarios. I'll defer to you deciding if you want to add this test

Reply:
Think I'm going to leave things for this PR.

This particular case I think the unit test coverage is likely enough, we probably don't need to verify the agent stop function is doing what it should. Covering the timer case outside of a unit test or the errored may add some benefit. I'll look into that separate from this PR so we aren't holding up a fix.

Reply:
works for me